### PR TITLE
Add registry mirror feature to snow

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clusterapi/config/containerd_config_append.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+    [plugins."io.containerd.grpc.v1.cri".registry.mirrors."public.ecr.aws"]
+    endpoint = ["https://{{.registryMirrorAddress}}"]
+    {{- if .registryCACert }}
```
Comment (vivek-koppuru) [inline pkg/clusterapi/config/containerd_config_append.toml:4]: Why is this red?

File: pkg/clusterapi/registryMirror.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		return "", fmt.Errorf("failed building containerd config file: %v", err)
+	}
+	return string(config), nil
+}
+
+func registryMirrorConfig(registryMirrorConfig *v1alpha1.RegistryMirrorConfiguration) (files []bootstrapv1.File, preKubeadmCommands []string, err error) {
+	registryAddress := net.JoinHostPort(registryMirrorConfig.Endpoint, registryMirrorConfig.Port)
+	registryConfig, err := registryMirrorConfigContent(registryAddress, registryMirrorConfig.CACertContent)
+	if err != nil {
+		return
```
Comment (vivek-koppuru) [inline pkg/clusterapi/registryMirror.go:36]: I know this works but why not explicitly return the values in this method instead?

File: pkg/clusterapi/config/containerd_config_append.toml (hunk 1 above)
Comment (jiayiwang7) [reply pkg/clusterapi/config/containerd_config_append.toml:4]: not sure, guess `toml` file type does not detect template format

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

File: pkg/clusterapi/config/containerd_config_append.toml (hunk 1 above)
Comment (jiayiwang7) [reply pkg/clusterapi/config/containerd_config_append.toml]: indentation issue, updated

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

File: pkg/clusterapi/config/containerd_config_append.toml (hunk 1 above)
Comment (jiayiwang7) [reply pkg/clusterapi/config/containerd_config_append.toml]: nvm idk maybe just file type highlight

**Pull request merged**